### PR TITLE
Adds detection for MacBook Air, MacBook Pro devices

### PR DESCRIPTION
**Pull request by liviuconcioiu**

### CODE REVIEW

**Comment by liviuconcioiu** [comment]
One thing I have doubts about are user agents starting with Safari.
`Safari/8536.30.1 CFNetwork/596.5 Darwin/12.5.0 (x86_64) (MacBookAir6%2C1)`
For example here, the Webkit version is 536.30.1, which is not detected right now. Also, number right before Webkit version isn't the Safari version.

**Comment by sanchezzzhak** [comment]
> One thing I have doubts about are user agents starting with Safari.
> `Safari/8536.30.1 CFNetwork/596.5 Darwin/12.5.0 (x86_64) (MacBookAir6%2C1)`
> For example here, the Webkit version is 536.30.1, which is not detected right now. Also, number right before Webkit version isn't the Safari version.

CFNetwork/<version>
it is not a version of webkit or other applications https://developer.apple.com/documentation/cfnetwork
```
Safari/10602.2.14.0.7 CFNetwork/720.5.7 Darwin/14.5.0 (x86_64)
Safari/10603.3.8 CFNetwork/720.5.7 Darwin/14.5.0 (x86_64)
Safari/11602.1.50.0.10 CFNetwork/760.6.3 Darwin/15.6.0 (x86_64)
Safari/11602.3.12.0.1 CFNetwork/760.6.3 Darwin/15.6.0 (x86_64)
Safari/11602.4.8.0.1 CFNetwork/760.6.3 Darwin/15.6.0 (x86_64)
Safari/11603.1.30.0.34 CFNetwork/760.6.3 Darwin/15.6.0 (x86_64)
Safari/11603.2.5 CFNetwork/760.6.5 Darwin/15.6.0 (x86_64)
Safari/11603.3.8 CFNetwork/760.6.5 Darwin/15.6.0 (x86_64)
Safari/11604.1.38.1.7 CFNetwork/760.6.5 Darwin/15.6.0 (x86_64)
Safari/12602.2.14.0.7 CFNetwork/807.1.3 Darwin/16.1.0 (x86_64)
Safari/12602.4.8 CFNetwork/807.2.14 Darwin/16.4.0 (x86_64)
Safari/12603.1.30.0.34 CFNetwork/811.4.18 Darwin/16.5.0 (x86_64)
Safari/12603.2.4 CFNetwork/811.5.4 Darwin/16.6.0 (x86_64)
Safari/12603.3.8 CFNetwork/811.5.4 Darwin/16.7.0 (x86_64)
Safari/12604.1.38.1.7 CFNetwork/811.5.4 Darwin/16.7.0 (x86_64)
Safari/12604.3.5.1.1 CFNetwork/811.5.4 Darwin/16.7.0 (x86_64)
Safari/12605.3.8.1 CFNetwork/811.9 Darwin/16.7.0 (x86_64)
Safari/13604.3.5 CFNetwork/889.9 Darwin/17.2.0 (x86_64)
Safari/13604.4.7.1.3 CFNetwork/893.13.1 Darwin/17.3.0 (x86_64)
Safari/13604.5.6 CFNetwork/893.13.1 Darwin/17.4.0 (x86_64)
Safari/13605.1.33.1.2 CFNetwork/897.15 Darwin/17.5.0 (x86_64)
Safari/13605.1.33.1.4 CFNetwork/897.15 Darwin/17.5.0 (x86_64)
Safari/13605.2.8 CFNetwork/901.1 Darwin/17.6.0 (x86_64)
Safari/13605.3.8 CFNetwork/902.1 Darwin/17.7.0 (x86_64)
Safari/13606.2.11 CFNetwork/902.1 Darwin/17.7.0 (x86_64)
Safari/14606.1.36.1.9 CFNetwork/974.1 Darwin/18.0.0 (x86_64)
```

**Comment by sanchezzzhak** [comment]
Safari/14606.1.36.1.9 CFNetwork/974.1 Darwin/18.0.0 (x86_64)
Safari/([0-9]{2})([0-9]{3})\.(\d+)\.(\d+)

safari $1.$3.$4 = 14.1.36    webkit $2 = 606
this is in theory, and even that won't be true

we know that the IOS version = version safari
we can assume that this request came from safari preview, then the operating system version will not match.

**Comment by liviuconcioiu** [comment]
I see

File: regexes/device/mobiles.yml [hunk 1]
```diff
@@ -562,6 +562,186 @@ Apple:
...
+      device: 'desktop'
+      model: 'MacBook Pro 15" (2012-2013)'
+
+    - regex: 'MacBookPro10(?:%2C|,)2'
+      device: 'desktop'
+      model: 'MacBook Pro 13" (2012-2013)'
+
+    - regex: 'MacBookPro11(?:%2C|,)1'
+      device: 'desktop'
+      model: 'MacBook Pro 13" (2013-2014)'
```
Comment (sanchezzzhak) [inline regexes/device/mobiles.yml:687]: I myself do not know whether to enter devices with two years of release with the same IDs.
I think it's better to leave the device with the very first mention by date.

`model: 'MacBook Pro 13" (2013)'` 

https://support.apple.com/HT201300

**Review by sanchezzzhak — CHANGES_REQUESTED** [review]

File: Tests/Parser/Client/fixtures/mobile_app.yml [hunk 2]
```diff
@@ -275,3 +275,45 @@
...
+    name: Microsoft Office Word
+    version: 16.11.319
+-
+  user_agent: Microsoft Office Excel/16.44.1213 (Mac OS/11.1; Desktop; en-US; AppStore; Apple/MacBookAir10,1)
+  client:
+    type: mobile app
+    name: Microsoft Office Excel
+    version: 16.44.1213
+-
+  user_agent: Microsoft Office OneNote/16.43/201106 (iOS/14.0.1; Tablet; es-MX; AppStore; Apple/iPad11,6)
```
Comment (sanchezzzhak) [inline Tests/Parser/Client/fixtures/mobile_app.yml:309]: here, too, you can define the device in the next PR

regex: `Apple/iPad11,6`
model: 'iPad 8 10.2" (WiFi)'

I think this list might help  https://everyi.com/by-identifier/ipod-iphone-ipad-specs-by-model-identifier.html

**Review by sanchezzzhak — DISMISSED** [review]

**Review by sanchezzzhak — DISMISSED** [review]

File: regexes/device/mobiles.yml [hunk 3]
```diff
@@ -562,6 +562,186 @@ Apple:
...
+      device: 'desktop'
+      model: 'MacBook Pro 13" (2012)'
+
+    - regex: 'MacBookPro11(?:%2C|,)1'
+      device: 'desktop'
+      model: 'MacBook Pro 13" (2013)'
+
+    - regex: 'MacBookPro11(?:%2C|,)[23]'
+      device: 'desktop'
+      model: 'MacBook Pro 15" (2013)'
```
Comment (sgiehl) [inline regexes/device/mobiles.yml:691]: According to https://support.apple.com/de-de/HT201300 those identifiers were used for models in 2013 and 2014. Seems that's also the case for some other models. Not sure if we should change the model name to `(2013/2014)` in such a case?

Comment (sanchezzzhak) [reply regexes/device/mobiles.yml:691]: If we specify a date range, we must accept this in the wiki agreement.

https://everymac.com/ultimate-mac-lookup/?search_keywords=MacBookPro11%2C3

2014-2015

Comment (sgiehl) [reply regexes/device/mobiles.yml:691]: @sanchezzzhak we can do that for sure. Not sure how to handle that the best way....

Comment (sanchezzzhak) [reply regexes/device/mobiles.yml:691]: I think we need a few more opinions to make a decision based on a majority of yes votes.

Comment (liviuconcioiu) [reply regexes/device/mobiles.yml:691]: My opinion is that `2013/2014` can be used only for two consecutive years, while `2013-2015` can be used for multiple years.

Comment (sgiehl) [reply regexes/device/mobiles.yml:691]: `2013-2015` is fine for me as well. Would you update the PR accordingly?

**Review by sgiehl — DISMISSED** [review]
Only had a rough look through the changes, but it imho looks good now.
@sanchezzzhak feel free to have another look and merge if everything is fine for you as well

**Review by sanchezzzhak — APPROVED** [review]

**Pull request merged**